src/core/pointcloud/qgspointcloudlayerrenderer.cpp

```diff
@@ -203,15 +229,92 @@ bool QgsPointCloudLayerRenderer::render()
       mReadyToCompose = true;
     }
   }
+  return nodesDrawn;
+}
 
-  QgsDebugMsgLevel( QStringLiteral( "totals: %1 nodes | %2 points | %3ms" ).arg( nodesDrawn )
-                    .arg( context.pointsRendered() )
-                    .arg( t.elapsed() ), 2 );
+int QgsPointCloudLayerRenderer::renderNodesAsync( const QVector<IndexedPointCloudNode> &nodes, QgsPointCloudIndex *pc, QgsPointCloudRenderContext &context, QgsPointCloudRequest &request, bool &canceled )
+{
+  int nodesDrawn = 0;
 
-  mRenderer->stopRender( context );
+  QElapsedTimer downloadTimer;
+  downloadTimer.start();
 
-  mReadyToCompose = true;
-  return !canceled;
+  const int groupSize = 4;
```

Comment:
I am not sure if I understand why we are doing groups here - it would be worth adding some comments to explain that... and is that even needed? It complicates the code...

Reply:
The previous initial implementation was to load all blocks in parallel and then render them sequentially. So I adjusted the implementation to load groupSize blocks in parallel and then render these groupSize blocks sequentially. It is needed since we might end up loading too many blocks which will make QGIS look non responsive when it's loading the blocks.
I will add an explanation, Do you think 4 is a good groupSize or should I change it to 8 or 16 ?